pkg/awsclient/client.go

```diff
@@ -0,0 +1,258 @@
+func GetAWSClient(kubeClient kubeclientpkg.Client, input NewAwsClientInput) (*awsClient, error) {
+
+	// error if region is not included
+	if input.AwsRegion == "" {
+		return nil, fmt.Errorf("getAWSClient:NoRegion: %v", input.AwsRegion)
+	}
+
+	if input.SecretName != "" && input.NameSpace != "" {
+		secret := &corev1.Secret{}
+		err := kubeClient.Get(context.TODO(),
+			types.NamespacedName{
+				Name:      input.SecretName,
+				Namespace: input.NameSpace,
+			},
+			secret)
+		if err != nil {
+			return nil, err
+		}
+		accessKeyID, ok := secret.Data[awsCredsSecretIDKey]
+		if !ok {
+			return nil, fmt.Errorf("AWS credentials secret %v did not contain key %v",
+				input.SecretName, awsCredsSecretIDKey)
+		}
+		secretAccessKey, ok := secret.Data[awsCredsSecretAccessKey]
+		if !ok {
+			return nil, fmt.Errorf("AWS credentials secret %v did not contain key %v",
+				input.SecretName, awsCredsSecretAccessKey)
+		}
+
+		awsClient, err := NewClient(string(accessKeyID), string(secretAccessKey), input.AwsToken, input.AwsRegion)
+		if err != nil {
+			return nil, err
+		}
+		return awsClient, nil
+	}
+
+	if input.AwsCredsSecretIDKey == "" && input.AwsCredsSecretAccessKey != "" {
+		return nil, fmt.Errorf("getAWSClient: NoAwsCredentials or Secret %v", input)
+	}
+
+	awsClient, err := NewClient(input.AwsCredsSecretIDKey, input.AwsCredsSecretAccessKey, input.AwsToken, input.AwsRegion)
+	if err != nil {
+		return nil, err
+	}
+	return awsClient, nil
+}
+
+func (c *awsClient) ApplySecurityGroupsToLoadBalancer(i *elb.ApplySecurityGroupsToLoadBalancerInput) (*elb.ApplySecurityGroupsToLoadBalancerOutput, error) {
+	return c.elbClient.ApplySecurityGroupsToLoadBalancer(i)
+}
+
+func (c *awsClient) ConfigureHealthCheck(i *elb.ConfigureHealthCheckInput) (*elb.ConfigureHealthCheckOutput, error) {
+	return c.elbClient.ConfigureHealthCheck(i)
+}
+
+func (c *awsClient) CreateLoadBalancer(i *elb.CreateLoadBalancerInput) (*elb.CreateLoadBalancerOutput, error) {
+	return c.elbClient.CreateLoadBalancer(i)
+}
+
+func (c *awsClient) CreateLoadBalancerListeners(i *elb.CreateLoadBalancerListenersInput) (*elb.CreateLoadBalancerListenersOutput, error) {
+	return c.elbClient.CreateLoadBalancerListeners(i)
+}
+
+func (c *awsClient) DeleteLoadBalancerListeners(i *elb.DeleteLoadBalancerListenersInput) (*elb.DeleteLoadBalancerListenersOutput, error) {
+	return c.elbClient.DeleteLoadBalancerListeners(i)
+}
+
+func (c *awsClient) DeregisterInstancesFromLoadBalancer(i *elb.DeregisterInstancesFromLoadBalancerInput) (*elb.DeregisterInstancesFromLoadBalancerOutput, error) {
+	return c.elbClient.DeregisterInstancesFromLoadBalancer(i)
+}
+func (c *awsClient) DescribeLoadBalancers(i *elb.DescribeLoadBalancersInput) (*elb.DescribeLoadBalancersOutput, error) {
+	return c.elbClient.DescribeLoadBalancers(i)
+}
+
+func (c *awsClient) DescribeLoadBalancersV2(i *elbv2.DescribeLoadBalancersInput) (*elbv2.DescribeLoadBalancersOutput, error) {
+	return c.elbv2Client.DescribeLoadBalancers(i)
+}
```

Comment:
SUPER unfortunate that these have the same upstream name. :( 👍 on your name for the elbv2 one.